app/src/NativeBase/Components/Checkbox.tsx

```diff
@@ -0,0 +1,44 @@
+/**
+ * @file CheckBox component for the STA Volunteer App
+ */
+
+import { Pressable } from 'native-base'
+import React from 'react'
+import { Ionicons } from '@expo/vector-icons'
+
+type CheckboxProps = {
+  onChange: (value: string[]) => void
+  checked: boolean
+}
+
+/**
+ * Component that displays a checkbox that can be checked or unchecked.
+ *
+ * @param { CheckboxProps} props The component props
+ * @param {value: string[] => void} props.onChange function called when the checkbox is pressed
+ * @param {boolean} props.checked whether the checkbox is checked or not
+ * @returns {React.ReactElement} Component
+ */
+
+const Checkbox: React.FC<CheckboxProps> = ({ onChange, checked }) => {
+  return (
+    <Pressable
+      onPress={() => onChange([])}
+      flexDirection={'row'}
+      alignItems={'center'}
+      borderColor={'purple.100'}
+      borderWidth={2}
+      borderRadius={2}
+      width={30}
+      height={30}
+      justifyContent={'center'}
+      _dark={{
+        borderColor: 'white',
+      }}
+    >
+      {checked && <Ionicons name="checkmark" size={20} color="#D1338A" />}
```

Comment:
Please always use colours from the theme rather than putting hex values directly into a component -- this makes it much easier to keep consistency and make updates later on.

If it won't let you reference the colour directly here as `color="magenta.100"` you can import the StaTheme file and reference the colour value from there (see e.g. [here](https://github.com/Scottish-Tech-Army/Volunteer-app/blob/8ee48822ba6fa2edbf9b57fcef33c21a39e78b46/app/src/NativeBase/Components/TopOfApp.tsx#L55))

Reply:
Also would it be possible to use the NativeBase `<Icon />` component here, like we usually do for icons, or was there a reason not to?  https://github.com/Scottish-Tech-Army/Volunteer-app/blob/develop/APP_DEVELOPMENT.md#icons

Reply:
Thanks . I wasn't aware of this Icon component. Will look into.

Reply:
I have added in the Material Icon with checkbox 'tick'. I have imported the StaTheme file to utilise the 'magenta.100' colour.